stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/header/MessageListHeader.kt

```diff
@@ -38,42 +38,54 @@ import io.getstream.chat.android.compose.ui.util.getDisplayName
  * This allows the user to fully govern how the [MessageListHeader] behaves, by passing in all the
  * data that's required to display it and drive its actions.
  *
- * @param modifier Modifier for styling.
  * @param channel Channel info to display.
  * @param currentUser The current user, required for different UI states.
- * @param isNetworkAvailable A flag that governs if we show the subtitle or the network loading view.
+ * @param modifier Modifier for styling.
  * @param messageMode The current message mode, that changes the header content, if we're in a Thread.
+ * @param isNetworkAvailable A flag that governs if we show the subtitle or the network loading view.
  * @param onBackPressed Handler that propagates the back button click event.
  * @param onHeaderActionClick Action handler when the user taps on the header action.
+ * @param leadingContent The content shown at the start of the header, by default a [BackButton].
+ * @param titleContent The content shown in the middle of the header and represents the core information, by default
+ * [DefaultMessageHeaderTitle].
+ * @param trailingContent The content shown at the end of the header, by default a [ChannelAvatar].
  */
 @Composable
 public fun MessageListHeader(
     channel: Channel,
     currentUser: User?,
-    isNetworkAvailable: Boolean,
-    messageMode: MessageMode,
-    onBackPressed: () -> Unit,
-    onHeaderActionClick: () -> Unit,
     modifier: Modifier = Modifier,
-) {
-
-    val memberCount = channel.memberCount
-    val onlineCount = channel.members.count { it.user.online }
-
-    val channelName = channel.getDisplayName()
-
-    val title = if (messageMode == Normal) {
-        channelName
-    } else {
-        stringResource(id = R.string.stream_compose_thread_title)
-    }
-
-    val subtitle = if (messageMode == Normal) {
-        stringResource(id = R.string.stream_compose_channel_members, memberCount, onlineCount)
-    } else {
-        stringResource(id = R.string.stream_compose_thread_subtitle, channelName)
-    }
+    messageMode: MessageMode = Normal,
+    isNetworkAvailable: Boolean = true,
+    onBackPressed: () -> Unit = {},
+    onHeaderActionClick: (Channel) -> Unit = {},
+    leadingContent: @Composable RowScope.() -> Unit = {
+        BackButton(
+            modifier = Modifier
+                .size(36.dp),
+            imageVector = Icons.Default.ArrowBack,
+            onBackPressed = onBackPressed,
+        )
+    },
 
+    titleContent: @Composable RowScope.() -> Unit = {
+        DefaultMessageHeaderTitle(
+            modifier = Modifier.weight(6f),
```

Comment:
I'm curious, does `6f` matter here as the value? Would `1f` have the same effect?

Reply:
That's a good point - `1f` works too as it fills the remaining space.